code/Simulation/SimulationAgent.py

```diff
@@ -0,0 +1,73 @@
+import asyncio
+from random import sample, randint
+
+from spade.agent import Agent
+from spade.behaviour import CyclicBehaviour
+from spade.message import Message
+
+from Simulation.Knowledge import Knowledge, KnowledgeInformation
+from Simulation.InformationSource import InformationSource
+
+
+def prepare_gossip_message(receiver, gossip):
+    msg = Message(to=receiver)
+    msg.body = gossip
+    return msg
+
+
+class SimulationAgent(Agent):
+
+    class PropagateGossipBehaviour(CyclicBehaviour):
+        async def run(self):
+            if len(self.agent.neighbours) == 0:
+                await asyncio.sleep(100)
+
+            information = self.agent.knowledge.get_random_information()
+
+            if information is not None:
+                receiver = sample(self.agent.neighbours, 1)[0]
+                message = prepare_gossip_message(receiver, information.body)
+                await self.send(message)
+                print("{}: I send message to {}".format(self.agent.jid, receiver))
```

Comment:
Czy nie powinno się uzupełnić tego brancha o logowanie, jak już będzie to na masterze?

Reply:
Myślę, że logowanie można dodać później, a tak już będzie jakaś działająca część na masterze, od której się będzie mozna odbić i robić dalsze ficzery